src/cpu.c

```diff
@@ -406,6 +412,25 @@ static cpu_state_t *get_cpu_state (size_t cpu_num, size_t state) /* {{{ */
 	return (&cpu_states[index]);
 } /* }}} cpu_state_t *get_cpu_state */
 
+#if defined(HAVE_PERFSTAT) /* {{{ */
+/* populate global aggregate cpu rate */
+static int total_rate(gauge_t *sum_by_state, size_t state, derive_t d,
+				          value_to_rate_state_t* conv, cdtime_t now)
+{
+	gauge_t rate = NAN;
+	value_t val = { .derive = d };
+	int status = value_to_rate (&rate, val, DS_TYPE_DERIVE, now, conv);
```

Comment:
Please inline the conversion to `value_t` to avoid the temporary variable:

```c
int status = value_to_rate (&rate, (value_t) { .derive = d }, DS_TYPE_DERIVE, now, conv);
```